backend: Race detector finds a couple of issues

## Describe the bug

Whilst testing the websocket multiplexer I noticed there were some issues the race golang detector uncovered in the cache module in the backend.

## To Reproduce
Steps to reproduce the bug:
1. `cd backend && go test -race -v -p 1 ./...`



```
==================
WARNING: DATA RACE
Read at 0x00c000296210 by goroutine 21:
  runtime.mapiterinit()
      /usr/local/go/src/runtime/map.go:877 +0x0
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.(*cache[go.shape.interface {}]).cleanUp()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:129 +0x197
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.New[go.shape.interface {}].gowrap1()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:46 +0x44

Previous write at 0x00c000296210 by goroutine 20:
  runtime.mapassign_faststr()
      /usr/local/go/src/runtime/map_faststr.go:223 +0x0
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.(*cache[go.shape.interface {}]).SetWithTTL()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:66 +0x1a9
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.(*cache[go.shape.interface {}]).Set()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:53 +0xa4
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.(*cache[interface {}]).Set()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:52 +0x1f
  github.com/headlamp-k8s/headlamp/backend/pkg/plugins.PopulatePluginsCache()
      /home/rene/dev/headlamp/backend/pkg/plugins/plugins.go:282 +0x3b0
  github.com/headlamp-k8s/headlamp/backend/pkg/plugins_test.TestPopulatePluginsCache()
      /home/rene/dev/headlamp/backend/pkg/plugins/plugins_test.go:440 +0x5c
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1690 +0x226
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1743 +0x44

Goroutine 21 (running) created at:
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.New[go.shape.interface {}]()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:46 +0x1a5
  github.com/headlamp-k8s/headlamp/backend/pkg/plugins_test.TestPopulatePluginsCache()
      /home/rene/dev/headlamp/backend/pkg/plugins/plugins_test.go:437 +0x3a
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1690 +0x226
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1743 +0x44

Goroutine 20 (finished) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1743 +0x825
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:2168 +0x85
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1690 +0x226
  testing.runTests()
      /usr/local/go/src/testing/testing.go:2166 +0x8be
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:2034 +0xf17
  main.main()
      _testmain.go:59 +0x164
==================
==================
WARNING: DATA RACE
Read at 0x00c0002ac290 by goroutine 21:
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.(*cache[go.shape.interface {}]).cleanUp()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:129 +0x22e
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.New[go.shape.interface {}].gowrap1()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:46 +0x44

Previous write at 0x00c0002ac290 by goroutine 20:
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.(*cache[go.shape.interface {}]).SetWithTTL()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:66 +0x1ba
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.(*cache[go.shape.interface {}]).Set()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:53 +0xa4
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.(*cache[interface {}]).Set()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:52 +0x1f
  github.com/headlamp-k8s/headlamp/backend/pkg/plugins.PopulatePluginsCache()
      /home/rene/dev/headlamp/backend/pkg/plugins/plugins.go:282 +0x3b0
  github.com/headlamp-k8s/headlamp/backend/pkg/plugins_test.TestPopulatePluginsCache()
      /home/rene/dev/headlamp/backend/pkg/plugins/plugins_test.go:440 +0x5c
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1690 +0x226
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1743 +0x44

Goroutine 21 (running) created at:
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.New[go.shape.interface {}]()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:46 +0x1a5
  github.com/headlamp-k8s/headlamp/backend/pkg/plugins_test.TestPopulatePluginsCache()
      /home/rene/dev/headlamp/backend/pkg/plugins/plugins_test.go:437 +0x3a
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1690 +0x226
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1743 +0x44

Goroutine 20 (finished) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1743 +0x825
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:2168 +0x85
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1690 +0x226
  testing.runTests()
      /usr/local/go/src/testing/testing.go:2166 +0x8be
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:2034 +0xf17
  main.main()
      _testmain.go:59 +0x164
==================
==================
WARNING: DATA RACE
Read at 0x00c0002ac268 by goroutine 21:
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.(*cache[go.shape.interface {}]).cleanUp()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:129 +0x22e
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.New[go.shape.interface {}].gowrap1()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:46 +0x44

Previous write at 0x00c0002ac268 by goroutine 20:
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.(*cache[go.shape.interface {}]).SetWithTTL()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:66 +0x1ba
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.(*cache[go.shape.interface {}]).Set()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:53 +0xa4
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.(*cache[interface {}]).Set()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:52 +0x1f
  github.com/headlamp-k8s/headlamp/backend/pkg/plugins.PopulatePluginsCache()
      /home/rene/dev/headlamp/backend/pkg/plugins/plugins.go:268 +0x94
  github.com/headlamp-k8s/headlamp/backend/pkg/plugins_test.TestPopulatePluginsCache()
      /home/rene/dev/headlamp/backend/pkg/plugins/plugins_test.go:440 +0x5c
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1690 +0x226
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1743 +0x44

Goroutine 21 (running) created at:
  github.com/headlamp-k8s/headlamp/backend/pkg/cache.New[go.shape.interface {}]()
      /home/rene/dev/headlamp/backend/pkg/cache/cache.go:46 +0x1a5
  github.com/headlamp-k8s/headlamp/backend/pkg/plugins_test.TestPopulatePluginsCache()
      /home/rene/dev/headlamp/backend/pkg/plugins/plugins_test.go:437 +0x3a
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1690 +0x226
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1743 +0x44

Goroutine 20 (finished) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1743 +0x825
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:2168 +0x85
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1690 +0x226
  testing.runTests()
      /usr/local/go/src/testing/testing.go:2166 +0x8be
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:2034 +0xf17
  main.main()
      _testmain.go:59 +0x164
==================
    plugins_test.go:70: Got create file event in the sub directory
    plugins_test.go:79: Got delete file event in the sub directory
    testing.go:1399: race detected during execution of test
--- FAIL: TestWatch (10.00s)
```

--- a/backend/cmd/multiplexer_test.go
+++ b/backend/cmd/multiplexer_test.go
@@ -200,6 +200,7 @@ func TestUpdateStatus(t *testing.T) {
 	conn := &Connection{
 		Status: ConnectionStatus{},
 		Done:   make(chan struct{}),
+		mu:     sync.RWMutex{},
 	}
 
 	// Test different state transitions
@@ -211,7 +212,9 @@ func TestUpdateStatus(t *testing.T) {
 	}
 
 	for _, state := range states {
+		conn.mu.Lock()
 		conn.Status.State = state
+		conn.mu.Unlock()
 		assert.Equal(t, state, conn.Status.State)
 	}
 
@@ -222,15 +225,18 @@ func TestUpdateStatus(t *testing.T) {
 
 		go func(i int) {
 			defer wg.Done()
-
+			conn.mu.Lock()
 			state := states[i%len(states)]
 			conn.Status.State = state
+			conn.mu.Unlock()
 		}(i)
 	}
 	wg.Wait()
 
 	// Verify final state is valid
+	conn.mu.RLock()
 	assert.Contains(t, states, conn.Status.State)
+	conn.mu.RUnlock()
 }
 
 func TestMonitorConnection_Reconnect(t *testing.T) {

--- a/backend/pkg/cache/cache.go
+++ b/backend/pkg/cache/cache.go
@@ -126,13 +126,13 @@ func (c *cache[T]) cleanUp() {
 	for {
 		<-ticker.C
 
+		c.lock.Lock()
 		for key, value := range c.store {
 			if !value.expiresAt.IsZero() && value.expiresAt.Before(time.Now()) {
-				c.lock.Lock()
 				delete(c.store, key)
-				c.lock.Unlock()
 			}
 		}
+		c.lock.Unlock()
 	}
 }
 

